modules/broker-core/src/main/java/org/wso2/broker/core/Broker.java

```diff
@@ -77,7 +84,9 @@ public Broker(StartupContext startupContext) throws Exception {
                                                                                         BrokerConfiguration.class);
         DataSource dataSource = startupContext.getService(DataSource.class);
         StoreFactory storeFactory = new StoreFactory(dataSource, metricManager, configuration);
-        this.messagingEngine = new MessagingEngine(storeFactory, metricManager);
+        SharedMessageStore sharedMessageStore = storeFactory.getSharedMessageStore(32768, 1024);
```

Comment:
We shouldn't create the shared message store here and pass it to the constructors. We should use the store factory inside respective constructors to do that. If we have a requirement to share the same shared message store we should handle that inside the factory classes it self.